src/main/java/pacApp/pacSecurity/CustomCorsFilter.java

```diff
@@ -1,27 +1,41 @@
 package pacApp.pacSecurity;
 
+import org.springframework.beans.factory.annotation.Value;
+
 import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 
 public class CustomCorsFilter implements Filter {
 
+    @Value("${allow.origin}")
+    private String origin;
+
     @Override
     public void init(FilterConfig filterConfig) throws ServletException {
         //
     }
 
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
-        if (!(servletResponse instanceof HttpServletResponse)) {
+        if (!(servletRequest instanceof HttpServletRequest) || !(servletResponse instanceof HttpServletResponse)) {
+            filterChain.doFilter(servletRequest, servletResponse);
+            return;
+        }
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+
+        if (!request.getMethod().equalsIgnoreCase("OPTIONS")) {
+            filterChain.doFilter(servletRequest, servletResponse);
             return;
         }
 
         HttpServletResponse response = (HttpServletResponse) servletResponse;
-        response.setHeader("Access-Control-Allow-Origin", "*");
+        response.setHeader("Access-Control-Allow-Origin", this.origin);
         response.setHeader("Access-Control-Allow-Methods", "GET,POST,PUT,DELETE,OPTIONS");
         response.setHeader("Access-Control-Allow-Headers", "*");
-        //response.setHeader("Access-Control-Allow-Credentials", true);
+        response.setHeader("Access-Control-Allow-Credentials", "true");
```

Comment:
Is thsi required? locally and on the azure platform the domain is *.azurewebsites.net . So it is practically a similar domain?

Reply:
The Access-Control-Allow-Credentials header seems to be relevant in relation to CORS and the handling or saving of cookies by the browser. This should also cover the case if a web application adapts to `xhrFields` in order to receive credentials from the server. 
`xhrFields: {
  withCredentials: true
}`

[Spring RestController not setting cookies with response](https://stackoverflow.com/questions/45823950/spring-restcontroller-not-setting-cookies-with-response)
The domain in the Access-Control-Allow-Origin header can now be set and customized with the application.properties after the compilation. This covers issue #6.

Reply:
Okay fine with me. I only thought is was redundant since we already have the same domain. But with the corret control header this should still be fine